vault documentation: doc cleanup effort-batch5 (#16913)
hashicorp/vault: website/content/docs/internals/telemetry.mdx

# Telemetry

The Vault server process collects various runtime metrics about the performance of different libraries and subsystems. These metrics are aggregated on a ten-second interval and retained for one minute in memory. Telemetry from Vault must be stored in metrics aggregation software to monitor Vault and collect durable metrics.

To view the raw data, you must send a signal to the Vault process: on Unix-style operating systems, this is `USR1`, while on Windows, it is `BREAK`. When the Vault process receives this signal, it will dump the current telemetry information to the process's `stderr`.

This telemetry information can be used for debugging purposes and provides users with insights into Vault's runtime.

> Telemetry information can also be streamed directly from Vault to a range of metrics aggregation solutions as described in the [telemetry Stanza][telemetry-stanza] documentation.

Reviewer: ```suggestion
Telemetry information can also be streamed directly from Vault to a range of metrics aggregation solutions as described in the [telemetry stanza][telemetry-stanza] documentation.
```

You'll note that log entries are prefixed with the metric type as follows:

- **[C]** is a counter. Counters are cumulative metrics that are incremented when some event occurs, and resets at the end of reporting intervals. Vault retains counters and other metrics for one minute in-memory, so an [aggregation solution][telemetry-stanza] must be configured to see accurate and persistent counters over time.
- **[G]** is a gauge. Gauges provide measurements of current values.
- **[S]** is a summary. Summaries provide sample observations of values. Vault commonly uses summaries for measuring the timing duration of discrete events in the reporting interval.

The following sections describe the available Vault metrics. The metrics interval are approximately 10 seconds when manually triggering metrics output using the above-described signals. Some high-cardinality gauges, like `vault.kv.secret.count`, are emitted every 10 minutes, or at an interval configured in the `telemetry` stanza.

> Some Vault metrics come with additional [labels](#metric-labels) describing the measurement in more detail, such as the namespace in which an operation takes place or the auth method used to create a token. This additional information is incorporated into the metrics name in the in-memory telemetry or other telemetry engines that do not support labels. The metric name in the table bleow is followed by a list of labels supported, in the order in which they appear, if flattened.

Reviewer: ```suggestion
Some Vault metrics come with additional [labels](#metric-labels) describing the measurement in more detail, such as the namespace in which an operation takes place or the auth method used to create a token. This additional information is incorporated into the metrics name in the in-memory telemetry or other telemetry engines that do not support labels. The metric name in the table below is followed by a list of labels supported, in the order in which they appear, if flattened.
```

| :---------------------------------- | :---------------------------------------------------------------------------------------------------- | :---- | :---- |
| `vault.autopilot.node.healthy`      | Set to 1 if the node_id is deemed healthy by Autopilot, 0 if not                                      | bool  | gauge |
| `vault.autopilot.healthy`           | Set to 1 if Autopilot considers all nodes healthy                                                     | bool  | gauge |

> | `vault.autopilot.failure_tolerance` | How many nodes can be lost while maintaining quorum, i.e., number of healthy nodes more than quorum | nodes | gauge |

Reviewer: To explain failure tolerance, `in excess of` makes better sense, so you can leave it as-is.

```suggestion
| `vault.autopilot.failure_tolerance` | How many nodes can be lost while maintaining quorum, i.e., number of healthy nodes in excess of quorum | nodes | gauge |
```

| `vault.raft.state.candidate`    | Increments whenever raft server starts an election                                                            | Elections | counter |
| `vault.raft.state.leader`       | Increments whenever raft server becomes a leader                                                              | Leaders   | counter |

> **Why are they vital?**: Raft cluster a stable leader during normal operations. If frequent elections or leadership changes occur, network issues between the raft nodes may be present or that the raft servers

Author: ```suggestion
**Why are they vital?**: Raft cluster should have a stable leader during normal operations. If frequent elections or leadership changes occur, network issues between the raft nodes may be present or the raft servers
```

> **Why are they vital?**: Raft cluster should have a stable leader during normal operations. If frequent elections or leadership changes occur, network issues between the raft nodes may be present or the raft servers
> cannot keep up with the load.

Reviewer: ```suggestion
**Why are they vital?**: If frequent elections or leadership changes occur, it would likely indicate network issues between the raft nodes or the raft servers cannot keep up with the load.
```
